radis/test/lbl/test_broadening.py

```diff
@@ -938,6 +938,57 @@ def test_noneq_continuum(plot=False, verbose=2, warnings=True, *args, **kwargs):
     assert res < 5.2e-6
 
 
+def test_broadening_chunksize_eq(verbose=True, plot=False, *args, **kwargs):
+    """
+    Test an equilibrium spectrum with and without chunksize,
+    using different optimization strategies
+    and ensure that the residual is very small.
+
+    """
+    sf = SpectrumFactory(
+        wavelength_min=4000,
+        wavelength_max=4500,
+        cutoff=1e-27,
+        pressure=1,
+        isotope="1,2",
+        truncation=5,
+        neighbour_lines=5,
+        path_length=0.1,
+        mole_fraction=1e-3,
+        medium="vacuum",
+        optimization=None,
+        chunksize=None,
+        wstep=0.001,
+        verbose=False,
+    )
+    sf.load_databank("HITEMP-CO")
+    sf.params.broadening_method = "convolve"
+    Tgas = 2000
+    chunk_loop = 1000000
+    plot_diff_list = []
+    # Testing all 6 cases of chunksize with optimization
+    for chunksize in [None, chunk_loop]:
+        sf.misc["chunksize"] = chunksize  # Setting chunksize
+        s_new = sf.eq_spectrum(Tgas=Tgas)
+
+        for optimization in [None, "simple", "min-RMS"]:
+            sf.params["optimization"] = optimization  # Setting optimization
+            s = sf.eq_spectrum(Tgas=Tgas)
+            # Residual calculation
+            res = get_residual(s, s_new, "abscoeff")
```

Comment:
/!\ Warning.

You are not testing Chunksize=None Vs chunksize!= None here, are you ? 

Seems both s_new and s are computed with the same chunksize parameter.

Can you check this, and also use more specific names for spectra (like s_no_chunksize; s_chunksize)

Reply:
Hello, I tested this separately on my system, and the chunksize param being printed is correct

Reply:
You do not compare Chunksize vs No-chunksize in this loop. 
s_new and s have the same chunksize value.

Reply:
Okay, I'll fix it. Thanks!

Reply:
Hello @erwanp, can you check? I think it should work now.

Reply:
Looks good